src/storage/DataAccessorBasedStore.ts

```diff
@@ -618,9 +631,15 @@ export class DataAccessorBasedStore implements ResourceStore {
 
     // Create the container, starting with its parent
     const ancestors = this.identifierStrategy.isRootContainer(container) ?
-      [] :
+      {} :
       await this.createRecursiveContainers(this.identifierStrategy.getParentContainer(container));
-    await this.writeData(container, new BasicRepresentation([], container), true, false, false);
-    return [ ...ancestors, container ];
+    const changes = await this.writeData(container, new BasicRepresentation([], container), true, false, false);
+    const filteredChanges = Object.entries(changes).reduce(
+      (acc, [ key, value ]): Record<string, RepresentationMetadata> =>
+        value.get(SOLID_AS.terms.Activity)?.value === AS.Create ? { ...acc, [key]: value } : acc,
+      {},
+    );
+
+    return { ...ancestors, ...filteredChanges };
```

Comment:
I'm not 100% sure what you're trying to achieve here (so adding a comment would be helpful to make sure).

You only keep the resources that got a create activity added in `writeData`, but I'm not sure why. To prevent having an update entry instead of a create entry on certain parent containers? I think just doing `return { ...changes, ...ancestors }` would fix that, as any entry in both changes and ancestors would use the ancestors entry, which will have the create activity.

This way you also have the update activity for the "root parent" (parent of the first container that gets created) which I think is currently missing.